node/src/types/deploy.rs

```diff
@@ -1615,6 +1615,53 @@ pub struct DeployMetadata {
     pub execution_results: HashMap<BlockHash, ExecutionResult>,
 }
 
+/// Additional information describing a deploy.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub enum DeployMetadataExt {
+    /// Holds the execution results of a deploy.
+    Metadata(DeployMetadata),
+    /// Holds the hash and height of the block this deploy was included in.
+    BlockInfo(BlockHashAndHeight),
+    /// No execution results or block information available.
+    Empty,
+}
```

Comment:
Couldn't we just get away with adding more fields to `DeployMetadata` struct? Or put `execution_results` directly in `DeployMetadataExt`? What we have now is a nested metadata :)

Another alternative could be to rename the `DeployMetadata` to `ExecutionResults` :thinking:

Reply:
We wanted the solution here to be backwards compatible. `DeployMetadata` is used in many other places and, with this in mind, we can do no renames in `DeployMetadata`. As `DeployMetadataExt` is a completely new struct, we could just add execution results into it, but the point was that we only care about block hash and height when we don't have the execution results, and vice versa, so it made sense to have it as an `enum`, with either or. This also makes it easier to work with the (unlikely) case where we have neither.

The point about nested metadata is valid, but I think it's because `DeployMetadata` is not the best name for what is basically just execution results.

Considering all this, if you don't have strong opinions here, I'd like to keep it as is. We will have flexibility with `DeployMetadataExt` in the future as it's only used for this RPC query and we fully control how it looks like to the end user.

Reply:
Thanks for explanation :+1: